bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/StateProfileTypeImpl.java

```diff
@@ -52,4 +55,9 @@ public String getLabel() {
         return label;
     }
 
+    @Override
+    public Collection<String> getSupportedItemTypesOfChannel() {
+        return supportedItemTypesOfChannel;
```

Comment:
The caller is able to modify the collection.
I don't think this is the desired behavior. Is it?
Why not holding a unmodifiable view only, so no need to act here but once in the constructor only (perhaps copy the provided collection once depend on the builder impl.).